From: Martin Schwidefsky <schwidefsky@de.ibm.com>
Subject: s390/sclp: single increment assignment control
Patch-mainline: v4.14-rc1
Git-commit: 45be0a02f8110a13502fac6167a893c7c0eff432
References: FATE#323299, bsc#1061050

Summary:     kernel: single increment assignment of memory 
Description: Enable a new option of the Attach-Storage-Element SCLP
             command to speed up memory hotplug in the case that the
             memory hotplug operation requires the attachment of a
             new storage element.

Upstream-Description:

             s390/sclp: single increment assignment control

             Set a new option bit of the attach command to speed up memory
             hotplug.

             Signed-off-by: Martin Schwidefsky <schwidefsky@de.ibm.com>


Signed-off-by: Martin Schwidefsky <schwidefsky@de.ibm.com>
Acked-by: Hannes Reinecke <hare@suse.com>
---
 drivers/s390/char/sclp_cmd.c |    1 +
 1 file changed, 1 insertion(+)

--- a/drivers/s390/char/sclp_cmd.c
+++ b/drivers/s390/char/sclp_cmd.c
@@ -252,6 +252,7 @@ static int sclp_attach_storage(u8 id)
 	if (!sccb)
 		return -ENOMEM;
 	sccb->header.length = PAGE_SIZE;
+	sccb->header.function_code = 0x40;
 	rc = sclp_sync_request_timeout(0x00080001 | id << 8, sccb,
 				       SCLP_QUEUE_INTERVAL);
 	if (rc)
